modules/frontend/querysharding.go

```diff
@@ -167,3 +154,41 @@ func DoRequests(ctx context.Context, downstream Handler, reqs []*http.Request) (
 
 	return resps, firstErr
 }
+
+func mergeResponses(rrs []RequestResponse) (*http.Response, error) {
+	var errCode = http.StatusOK
+	var errBody io.ReadCloser
+	var combinedTrace []byte
+	for _, rr := range rrs {
+		if rr.Response.StatusCode == http.StatusOK {
+			body, err := ioutil.ReadAll(rr.Response.Body)
```

Comment:
do we need a `rr.Response.Body.Close()` or does `iotuil.ReadAll` handle that?

Reply:
Nice catch!